platforms/documentation/docs/src/samples/writing-tasks/custom-test-task/common/plugin/src/main/java/com/example/CustomTest.java

```diff
@@ -0,0 +1,104 @@
+                    suite.started(Instant.now());
+                    try (TestEventReporter test = suite.reportTest("failingTest", "failingTest()")) {
+                        test.started(Instant.now());
+                        test.failed(Instant.now(), "This is a test failure");
+                    }
+                    suite.failed(Instant.now(), "This is additional message for the suite failure");
+                }
+                hasAnyFailures = true;
+            }
+
+            // Demonstrate multiple test suites with tests with different outcomes
+            Collection<String> suiteNames = Arrays.asList("MyTestSuite", "MyOtherTestSuite", "AnotherTestSuite");
+            for (String suiteName : suiteNames) {
+                try (GroupTestEventReporter suite = root.reportTestGroup(suiteName)) {
+                    suite.started(Instant.now());
+                    int testCount = random.nextInt(9) + 1;
+
+                    for (int i=0; i<testCount; i++) {
+                        try (TestEventReporter test = suite.reportTest("test" + i, "test(" + i + ")")) {
+                            test.started(Instant.now());
+
+                            test.output(Instant.now(), TestOutputEvent.Destination.StdOut, "This is some standard output");
+                            test.output(Instant.now(), TestOutputEvent.Destination.StdErr, "This is some standard error");
+
+                            // randomly skip some tests
+                            if (random.nextBoolean()) {
+                                test.skipped(Instant.now());
+                            } else {
+                                test.succeeded(Instant.now());
+                            }
+                        }
+                    }
+                    suite.succeeded(Instant.now());
+                }
+            }
+
+            // Demonstrate arbitrary nesting levels
+            try (GroupTestEventReporter outer = root.reportTestGroup("OuterNestingSuite")) {
+                outer.started(Instant.now());
+                try (GroupTestEventReporter deeper = outer.reportTestGroup("DeeperNestingSuite")) {
+                    deeper.started(Instant.now());
+                    try (GroupTestEventReporter inner = deeper.reportTestGroup("InnerNestingSuite")) {
+                        inner.started(Instant.now());
+                        try (TestEventReporter test = inner.reportTest("nestedTest", "nestedTest()")) {
+                            test.started(Instant.now());
+                            test.succeeded(Instant.now());
+                        }
+                        inner.succeeded(Instant.now());
+                    }
+                    deeper.succeeded(Instant.now());
+                }
+                outer.succeeded(Instant.now());
+            }
+
+            if (hasAnyFailures) {
+                root.failed(Instant.now());
+            } else {
+                root.succeeded(Instant.now());
```

Comment:
💭 Seeing `Instant.now()` repeated so many times really makes me think it's worth adding default overloads of these event reporting methods that automatically get the time from here without typing if right from the start.